test/org/sagebionetworks/bridge/hibernate/HibernateHelperTest.java

```diff
@@ -270,4 +261,180 @@ public void execute() {
+        verify(mockExceptionConverter).convert(ex, null);
+    }
+    
+    @Test
+    public void queryUpdateConvertsExceptions() throws Exception {
+        ConstraintViolationException cve = new ConstraintViolationException(
+                "Duplicate entry 'studyTest-email@email.com' for key 'Accounts-StudyId-Email-Index'", null, null);
+        PersistenceException ex = new PersistenceException(cve);
+        
+        reset(helper); // clear the doAnswer set up in @before 
+        // this isn't exactly when the exception is thrown, but it's close enough to simulate
+        doThrow(ex).when(mockSessionFactory).openSession();
+        
+        HibernateAccount account = new HibernateAccount();
+        account.setStudyId("testStudy");
+        
+        when(mockExceptionConverter.convert(ex, null)).thenReturn(TEST_EXCEPTION);
+        
+        try {
+            helper.queryUpdate("query string", ImmutableMap.of());
+            fail("Should have thrown exception");
+        } catch(Exception e) {
+            assertSame(TEST_EXCEPTION, e);
+        }
+        verify(mockExceptionConverter).convert(ex, null);
+    }
+    
+    @Test
+    public void updateConvertsExceptions() throws Exception {
+        ConstraintViolationException cve = new ConstraintViolationException(
+                "Duplicate entry 'studyTest-email@email.com' for key 'Accounts-StudyId-Email-Index'", null, null);
+        PersistenceException ex = new PersistenceException(cve);
+        
+        
+        reset(helper); // clear the doAnswer set up in @before 
+        // this isn't exactly when the exception is thrown, but it's close enough to simulate
+        doThrow(ex).when(mockSessionFactory).openSession();
+        
+        HibernateAccount account = new HibernateAccount();
+        account.setStudyId("testStudy");
+
+        when(mockExceptionConverter.convert(ex, account)).thenReturn(TEST_EXCEPTION);
+
+        try {
+            helper.update(account);
+            fail("Should have thrown exception");
+        } catch(Exception e) {
+            assertSame(TEST_EXCEPTION, e);
+        }
+        verify(mockExceptionConverter).convert(ex, account);
+    }
+    
+    @Test
+    public void converterReturnsTheOriginalException() throws Exception {
+        // This exception is not in the PersistenceException hierarchy and should be returned as is.
+        IllegalArgumentException iae = new IllegalArgumentException("Illegal argument.");
+        reset(helper); // clear the doAnswer set up in @before 
+        // this isn't exactly when the exception is thrown, but it's close enough to simulate
+        doThrow(iae).when(mockSessionFactory).openSession();
+        
+        HibernateAccount account = new HibernateAccount();
+        account.setStudyId("testStudy");
+        try {
+            helper.update(account);
+            fail("Should have thrown exception");
+        } catch(RuntimeException e) {
+            assertSame(iae, e);
+        }
```

Comment:
This test doesn't test what you think it does. executeWithExceptionHandling() only catches PersistenceExceptions, so the IllegalArgumentException is uncaught and being throw all the way up to the test, bypassing the code we're trying to test.